backend/estimate-service/src/main/java/org/egov/works/service/EnrichmentService.java

```diff
@@ -0,0 +1,156 @@
+package org.egov.works.service;
+
+import digit.models.coremodels.AuditDetails;
+import digit.models.coremodels.IdResponse;
+import org.apache.commons.lang3.StringUtils;
+import org.egov.common.contract.request.RequestInfo;
+import org.egov.common.contract.request.User;
+import org.egov.tracer.model.CustomException;
+import org.egov.works.config.EstimateServiceConfiguration;
+import org.egov.works.repository.EstimateRepository;
+import org.egov.works.repository.IdGenRepository;
+import org.egov.works.util.EstimateServiceUtil;
+import org.egov.works.web.models.Estimate;
+import org.egov.works.web.models.EstimateDetail;
+import org.egov.works.web.models.EstimateRequest;
+import org.egov.works.web.models.EstimateSearchCriteria;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.util.CollectionUtils;
+
+import java.math.BigDecimal;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static org.egov.works.util.EstimateServiceConstant.UPDATE_ROLES;
+
+@Service
+public class EnrichmentService {
+
+    @Autowired
+    private EstimateServiceUtil estimateServiceUtil;
+
+    @Autowired
+    private IdGenRepository idGenRepository;
+
+    @Autowired
+    private EstimateServiceConfiguration config;
+
+    @Autowired
+    private EstimateRepository estimateRepository;
+
+
+    public void enrichCreateEstimate(EstimateRequest request) {
+        RequestInfo requestInfo = request.getRequestInfo();
+        Estimate estimate = request.getEstimate();
+        List<EstimateDetail> estimateDetails = estimate.getEstimateDetails();
+
+        AuditDetails auditDetails = estimateServiceUtil.getAuditDetails(requestInfo.getUserInfo().getUuid(), estimate, true);
+        estimate.setAuditDetails(auditDetails);
+        estimate.setId(UUID.randomUUID());
+        Date currentDT = new Date();
+        BigDecimal proposalDate = new BigDecimal(currentDT.getTime());
+        estimate.setProposalDate(proposalDate);
+
+        String rootTenantId = estimate.getTenantId().split("\\.")[0];
+
+        List<String> estimateNumbers = getIdList(requestInfo, rootTenantId
+                , config.getIdgenEstimateNumberName(), config.getIdgenEstimateNumberFormat(), 1);
+
+        String estimateNumSeq = "";
+        if (estimateNumbers != null && !estimateNumbers.isEmpty()) {
+            String estimateNumber = estimateNumbers.get(0);
+            estimate.setEstimateNumber(estimateNumber);
+            String[] estimateNumberArr = estimateNumber.split("/");
+            estimateNumSeq = estimateNumberArr[estimateNumberArr.length - 1];
+        }
+
+        if (estimateDetails != null && !estimateDetails.isEmpty() && StringUtils.isNotBlank(estimateNumSeq)) {
+            List<String> estimateDetailNumbers = getIdList(requestInfo, rootTenantId
```

Comment:
Do we need rootTenantId? Try once with the original tenantId received in the request.

Reply:
As per my analysis, we need the same tenant Id that we have configured https://github.com/egovernments/works-mdms-data/blob/76a1b7db21a807dfa134e825fec7f2fb5ef11ed0/data/pb/common-masters/IdFormat.json#L2 .  In others services, rootTenantIId has been used.

If we are passing original tenant Id (e.g: 'pb.amritsar') then our case , we are getting 

` "Errors": [
        {
            "code": "400 BAD_REQUEST",
            "message": "No Format is available in the MDMS for the given name and tenant",
            "description": null,
            "params": {}
        }
    ]`